pkg/controllers/deploymentreplicassyncer/deployment_replicas_syncer_controller.go

```diff
@@ -57,6 +57,13 @@ var predicateFunc = predicate.Funcs{
 			return false
 		}
 
+		// if old deployment is not labeled `retain-replicas`, but new is labeled, reconcile is needed.
+		// in case of deployment status changed before `retain-replicas` labeled.
+		oldRetainReplicasLabel := util.GetLabelValue(oldObj.GetLabels(), util.RetainReplicasLabel)
+		if oldRetainReplicasLabel != util.RetainReplicasValue {
+			return true
+		}
+
```

Comment:
It seems that as long as `util.RetainReplicasLabel` changes, then it should be re-`reconcile`?

Reply:
yes, then get latest deployment, only if deployment's `spec.replicas` not equal to `status.replicas` would do the update.

Reply:
> It seems that as long as `util.RetainReplicasLabel` changes, then it should be re-`reconcile`?

To be precise, there was no such label before, but now there is, it will be reconciled.

Reply:
Got it.